GitCommands/CustomDiffMergeToolCache.cs

```diff
@@ -59,20 +56,31 @@ public async Task<IEnumerable<string>> GetToolsAsync(GitModule module, int delay
 
             // The command will compete with other resources, avoid delaying startup
             // Do not bother with cancel tokens
-            await Task.Delay(delay);
+            await Task.Delay(delay, cancellationToken);
+
+            await _mutex.WaitAsync(cancellationToken).ConfigureAwait(false);
 
-            await _mutex.WaitAsync().ConfigureAwait(false);
             try
             {
                 if (_tools is null)
                 {
                     await TaskScheduler.Default;
+                    cancellationToken.ThrowIfCancellationRequested();
+
                     var toolKey = _isDiff ? SettingKeyString.DiffToolKey : SettingKeyString.MergeToolKey;
                     var defaultTool = module.GetEffectiveSetting(toolKey);
-                    string output = module.GetCustomDiffMergeTools(_isDiff);
+                    string output = module.GetCustomDiffMergeTools(_isDiff, cancellationToken);
                     _tools = ParseCustomDiffMergeTool(output, defaultTool);
                 }
             }
+            catch
+            {
+                if (_tools is null)
+                {
+                    // Parsing has failed, just provide an empty list, no user notification
+                    _tools = new List<string>();
```

Comment:
Not accepted by the compiler , the correct System.Array is referenced
```
Error	CS0426	The type name 'Empty<>' does not exist in the type 'Array'	GitCommands	C:\dev\gc\gitextensions_4\GitCommands\CustomDiffMergeToolCache.cs	81	Active
```

Reply:
Is correct namespace referenced? https://docs.microsoft.com/dotnet/api/system.array.empty

Reply:
I pulled down your branch, and it compiles for me
![image](https://user-images.githubusercontent.com/4403806/130347284-b0a3b5e7-2f21-43ba-859f-15200523e3c1.png)

Reply:
Now it worked, no clue why it could not be compiled before

> I pulled down your branch, and it compiles for me

This not changed in this PR (compare to the separate commit with cleanup I added..)